x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -1101,6 +1103,15 @@ synchronized void handleFailure(Exception e) {
                     "task encountered irrecoverable failure: " + e.getMessage() :
                     "task encountered more than " + transformTask.getNumFailureRetries() + " failures; latest failure: " + e.getMessage();
                 failIndexer(failureMessage);
+            } else {
+                // Since our schedule fires again very quickly after failures it is possible to run into the same failure numerous
+                // times in a row, very quickly. We do not want to spam the audit log with repeated failures, so only record the first one
+                if (e.getMessage().equals(lastAuditedExceptionMessage) == false) {
+                    auditor.warning(transformTask.getTransformId(),
+                        "Data frame transform encountered an exception: " + e.getMessage() +
+                            " Will attempt again at next scheduled trigger.");
```

Comment:
nit: Do we need a dot after the message? It seems like 2 different sentences.

Reply:
@dimitris-athanasiou I waffled on that. But certain Messages already had periods, then it turned into `Data frame ... exception: Something went wrong.. Will attempt... trigger.` which looked more strange to me than just not having a period at all.